### ISSUE TITLE
Support LoggerFactory

### ISSUE THREAD

**Issue by lavrov** [typelevel/log4cats#224]
Abstract `LoggerFactory[F[_]]` which would provide implementation agnostic factory methods. Having such factory would allow one to choose implementation at one place once.

**Comment by lorandszakacs** [comment]
Agree that this would be useful. I am willing to bikeshed the naming though. `Factory` is way too java.

**Comment by sirmax** [comment]
Regarding the naming, how about `Loggers[F[_]]`? Its API can mimic that of `Slf4jLogger`
```
trait Loggers[F[_]] {
  def get: Logger[F]
  def getFromName(name: String): Logger[F]
  def getFromClass(clazz: Class[_]): Logger[F]

  def create: F[Logger[F]]
  def fromName(name: String): F[Logger[F]]
  def fromClass(clazz: Class[_]): F[Logger[F]]
}
```

As a side-note: perhaps, it would be pragmatic to return not a plain `Logger[F]` but a full-blown `SelfAwareStructuredLogger[F]`.

**Comment by ivan-klass** [comment]
Seems like https://github.com/typelevel/log4cats/pull/629 is related PR for this

**Comment by lorandszakacs** [comment]
Available starting with `2.3.0` and `1.7.0` respectively.
